### PR TITLE
Add `allowlist.yml` to where it is not being handled in securityadmin tool.

### LINKED ISSUE #2047: [BUG] Security admin does not handle allowlist.yml in all the places where it handles whitelist.yml
**What is the bug?**
Security admin tool should handle `allowlist.yml` in all the places where it handles `whitelist.yml`.

### PR DESCRIPTION
**Pull request by cliu123**
Signed-off-by: cliu123 <lc12251109@gmail.com>

### Description
[Describe what this change achieves]
* Category (Enhancement, New feature, Bug fix, Test fix, Refactoring, Maintenance, Documentation) Bug fix

### Issues Resolved
#2047

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by DarshitChanpura — APPROVED** [review]

File: src/main/java/org/opensearch/security/tools/SecurityAdmin.java [hunk 1]
```diff
@@ -1218,6 +1220,9 @@ private static int upload(RestHighLevelClient tc, String index, String cd, boole
         if (new File(cd+"audit.yml").exists()) {
             success = uploadFile(tc, cd + "audit.yml", index, "audit", legacy, resolveEnvVars) && success;
         }
+        if (new File(cd+"allowlist.yml").exists()) {
```
Comment (cwperks) [inline src/main/java/org/opensearch/security/tools/SecurityAdmin.java:1223]: Does this mean that `allowlist.yml` is optional, but `whitelist.yml` is required?

Comment (cliu123) [reply src/main/java/org/opensearch/security/tools/SecurityAdmin.java:1223]: Good question! `allowlist.yml` is an alternative to and will be a replacement of `whitelist.yml`, so it does not necessarily exist in a cluster.

Comment (cliu123) [reply src/main/java/org/opensearch/security/tools/SecurityAdmin.java:1223]: @cwperks Good question! `allowlist.yml` is an alternative to and will be a replacement of `whitelist.yml`, so it does not necessarily exist in a cluster.

Comment (cwperks) [reply src/main/java/org/opensearch/security/tools/SecurityAdmin.java:1223]: There's an existence check around `allowlist.yml`, but no check around `whitelist.yml` a few lines above.

```
success = uploadFile(tc, cd+"whitelist.yml", index, "whitelist", legacy, resolveEnvVars) && success;
 if (new File(cd+"allowlist.yml").exists()) {
    success = uploadFile(tc, cd + "allowlist.yml", index, "allowlist", legacy, resolveEnvVars) && success;
}
```

When I run securityadmin after removing `whitelist.yml`, but have a file called `allowlist.yml` I get the following error: 

```ERR: Seems ../../../config/opensearch-security/whitelist.yml is not in legacy format: java.io.FileNotFoundException: ../../../config/opensearch-security/whitelist.yml (No such file or directory)```

I was wondering if it made sense to check for the existence of either and fail if neither exists.

Comment (cliu123) [reply src/main/java/org/opensearch/security/tools/SecurityAdmin.java:1223]: `whitelist.yml` is required, so there's no need to check the existence.

**Review by cwperks — APPROVED** [review]

**Pull request merged**